lib/nats.js

```diff
@@ -66,12 +66,27 @@ var VERSION = '0.6.8',
     PONG_RESPONSE = 'PONG' + CR_LF,
 
     // Errors
-    BAD_SUBJECT = 'Subject must be supplied',
-    BAD_MSG = 'Message can\'t be a function',
-    BAD_REPLY = 'Reply can\'t be a function',
-    CONN_CLOSED = 'Connection closed',
+    BAD_SUBJECT = 'BAD_SUBJECT',
+    BAD_SUBJECT_MSG = 'Subject must be supplied',
```

Comment:
Just an idea, but maybe there could custom error types?  Wondering whether something like this could be done for example:
```js
BAD_SUBJECT = 'Subject must be supplied';
BAD_SUBJECT_MSG = 'Subject must be supplied'
BAD_SUBJECT_ERR = function BAD_SUBJECT_ERR(err){
    Error.captureStackTrace(this, this.constructor);    
    this.name = this.constructor.name;
    this.message = BAD_SUBJECT_MSG;
};
util.inherits(BAD_SUBJECT_ERR, Error);
```

That way it would be possible to catch errors per type for example:
```js
try {
    // ...throws custom error
} catch (e) {
    if (e instanceof BAD_SUBJECT_ERR) {
        console.log("- Error: ", e.name);
        console.log("- Error: ", e.message);
        console.log("- Stack: ", e.stack);
    }
    if (e instanceof Error) {
        console.log("= Error: ", e.name);
        console.log("= Error: ", e.message);
        console.log("= Stack: ", e.stack);
    }
}
```